cmake/onnxruntime_providers.cmake

```diff
@@ -482,6 +482,7 @@ if (onnxruntime_USE_ACL)
   source_group(TREE ${ONNXRUNTIME_ROOT}/core FILES ${onnxruntime_providers_acl_cc_srcs})
   add_library(onnxruntime_providers_acl ${onnxruntime_providers_acl_cc_srcs})
   onnxruntime_add_include_to_target(onnxruntime_providers_acl onnxruntime_common onnxruntime_framework onnx onnx_proto protobuf::libprotobuf)
+  target_link_libraries(onnxruntime_providers_acl -L$ENV{LD_LIBRARY_PATH})
```

Comment:
why using LD_LIBRARY_PATH? that's normally used at run time.

Reply:
We need to link the ACL so files.

Reply:
My point was that LD_LIBRARY_PATH should not be used for that purpose. 
We should allow user to specify the location of ACL libraries as build option.

Reply:
Even if you make user pass ACL library location, the user will still need to set LD_LIBRARY_PATH in order to use onnxruntime_perf_test otherwise get this error: "error while loading shared libraries: libarm_compute.so: cannot open shared object file: No such file or directory."

Reply:
yes, but LD_LIBRARY_PATH is used at runtime. it's not meant to be used during build time (which is what the cmake files are for)
at build time, the ACL install location can be anywhere and not set in the user's LD_LIBRARY_PATH
the binary could then be deployed to a target where LD_LIBRARY_PATH would need to be set.
besides, LD_LIBRARY_PATH env variable is specific to Linux/Unix.
Many other EP's have dependency on other libs. (e.g. CUDA, OpenVINO) etc.  
But you can grep through all our cmake files and you won't find a single reference to LD_LIBRARY_PATH